### ISSUE TITLE
[A11y] Fix playlist edit/create forms

### ISSUE THREAD

**Issue by Dananji** [avalonmediasystem/avalon#6353]
### Description

Playlist edit/create forms on the playlist pages have the following accessibility issues identified from the accessibility check performed in https://github.com/avalonmediasystem/avalon/issues/6312;

| Success criterion | WCAG 2.1 Level | Accessibility checker | Error | Relevant HTML element |
| --- | --- | --- | --- | --- |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE | Form field missing a label | `<input>` textbox for `Name` |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE | Form field missing a label | `<span>` combobox for `Tags` |
| 4.1.2 Name, Role, Value | Level A | SiteImprove & WAVE | Form field missing a label | `<input>` textbox for `Tags` |

Since these form fields already have relevant form labels, we can change HTML to label/form field association in the DOM structure. 
Notes: https://www.a11yonrails.com/pages/forms.html

Both create playlist and edit playlist uses the same view template for the form: https://github.com/avalonmediasystem/avalon/blob/main/app/views/playlists/_form.html.erb

### Done Looks Like
- [ ] Accessibility checks pass for form input fields
